### PR TITLE
[explore] convert query and save btns to react

### PR DESCRIPTION
**Pull request by ascott**
first pass at moving explore.js from jquery to react. moves query and save btns to react. modal is still using the dom api. will move the modal to react in a subsequent pr.

plz review @mistercrunch @georgeke

### CODE REVIEW

File: caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <button type="button" className="btn btn-primary" onClick={this.props.onQuery}>
+          <i className="fa fa-bolt"></i>Query
+        </button>
+        <button type="button"
+                className={saveClasses}
+                onClick={this.props.onSave}
+                data-target="#save_modal"
+                data-toggle="modal"
+        >
+          <i className="fa fa-plus-circle"></i>Save as
```
Comment (ascott) [inline caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx]: need to figure out a pattern for getting translated strings into js.

Comment (ascott) [inline caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx]: could use http://airbnb.io/polyglot.js/...

**Comment by mistercrunch** [comment]
The travis build failed. You can see it's logs for details.

File: caravel/assets/javascripts/explore.js [hunk 2]
```diff
@@ -334,9 +338,14 @@ function initExploreView() {
     add_filter(undefined, "having");
   });
 
-  $(".query").click(function () {
-    query(true);
-  });
+  const queryAndSaveBtnsEl = document.getElementById('js-query-and-save-btns');
+  ReactDOM.render(
+    <QueryAndSaveBtns
```
Comment (mistercrunch) [inline caravel/assets/javascripts/explore.js]: this the line that failed, I think webpack needs a `.jsx` extension to get the right loader.

File: caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx (hunk 1 above)
Comment (mistercrunch) [inline caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx]: We use Python's Babel on the backend, hopefully we can have a consistent take on translations where all strings to be translated are all in the same framework.
http://babel.pocoo.org/en/latest/

Comment (ascott) [inline caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx]: right, we should definitely continue to use babel. i should write a helper so we have a consistent way to pass translated text from the server to bootstrap the ui.

Comment (ascott) [inline caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx]: something like:

server: 
```python
bootstrap_data('phrase_bundle_name', {
  phrase_key: 'translated text for key',
})
```

client:
```js
BootstrapData.get('phrase_bundle_name');
```

**Comment by mistercrunch** [comment]
It passed the build now, needs rebase

**Comment by georgeke** [comment]
Would it make more sense for `explore.jsx` to be in the explore/ directory as well? That's what I'm doing for `dashboard.jsx`

**Comment by georgeke** [comment]
LGTM once we pass the $('#btn_modal_save').click function as the onSave prop.

**Comment by ascott** [comment]
thanks for the review @georgeke. good suggestion on moving explore.jsx to the explore folder too.


the `$('#btn_modal_save')` element is referring to the save button in the modal.

screenshot 2016-07-07 18 11 54

the save button that i'm refactoring here triggers opening the modal, which is happening via the bootstrap modal dom api on these lines:
https://github.com/ascott/caravel/blob/master/caravel/assets/javascripts/explore/components/QueryAndSaveBtns.jsx#L23

we will definitely want to refactor the modal as well in another PR.

**Pull request merged**